docs/04-operation-guides/troubleshooting/eventing/evnt-05-fix-pending-messages.md

```diff
@@ -0,0 +1,137 @@
+---
+title: Published events are pending in the stream
+---
+
+## Symptom
+
+You publish events, but some of them are not received by the subscriber and stay pending in the stream.
+
+## Cause
+
+When NATS EventingBackend has more than 1 replica and the `Clustering` property on the NATS Server is enabled,
+a leader-election is taking place on the stream and consumer levels (
+see [NATS Documentation](https://docs.nats.io/running-a-nats-service/configuration/clustering/jetstream_clustering)).
+Once the leader is elected, all the messages are being replicated across the replicas.
+
+Sometimes replicas can go out-of-sync with the other replicas.
+As a result of this, messages on some consumers can stop being acknowledged and start piling up in the stream.
+
+## Remedy
+
+There are two ways of how to fix the "broken" consumers with pending messages. You will need either to trigger a leader
+reelection either on the consumers
+with pending messages or on the stream level.
+
+### Trigger Consumer leader election
+
+First, you need to find out which consumer(s) have pending messages. For that you need the latest version of NATS cli
+installed on your machine.
+You can find the broken consumer in two ways: by using Grafana dashboard or directly by using the NATS cli command.
+
+#### Find the broken consumers using Grafana dashboard
+
+1. [Access and Expose Grafana](../../security/sec-06-access-expose-grafana.md)
+2. Find the NATS JetStream Dashboard and check the pending messages
+   ![Pending consumer](../../assets/grafana_pending_consumer.png)
+3. Find the consumer with pending messages and encode it as md5 hash:
+
+```bash
+echo -n "tunas-testing/test-noapp3/kyma.sap.kyma.custom.noapp.order.created.v1" | md5
+```
+
+this shell command results in `6642d54a92f357ba28280b9cb609e79d`.
+
+4. Then, you need to find consumer's leader:
+
+```bash
+nats consumer info sap 6642d54a92f357ba28280b9cb609e79d
+```
+
+```bash
+Information for Consumer sap > 6642d54a92f357ba28280b9cb609e79d created 2022-10-24T15:49:43+02:00
+
+Configuration:
+
+                Name: 6642d54a92f357ba28280b9cb609e79d
+         Description: tunas-testing/test-noapp3/kyma.sap.kyma.custom.noapp.order.created.v1
+          ...
+
+Cluster Information:
+
+                Name: eventing-nats
+              Leader: eventing-nats-1 # that's what we need
+             Replica: eventing-nats-0, current, seen 0.96s ago
+             Replica: eventing-nats-2, current, seen 0.96s ago
+```
+
+You can see, that its leader is the `eventing-nats-1` replica.
+
+#### Find the broken consumers using the NATS cli
+
+If you have NATS cli installed on your machine, you can simply run this shell script:
+
+   ```bash
+   for consumer in $(nats consumer list -n sap) # sap is the stream name
+   do
+     nats consumer info sap $consumer -j | jq -c '{name: .name, pending: .num_pending, leader: .cluster.leader}'
+   done
+   ```
+
+this will output the following:
```

Comment:
```suggestion
You must get the following output:
```

Reply:
I am really not sure, if `must` is the right word to use in a troubleshooting guide. I think it is too strict. What do you think about `have to` or `should` as alternatives ?

Reply:
According to [our guidelines](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/02-style-and-terminology.md#terminology) `must` is the best choice. 
If we go with `should` it'd mean that there is another possibility. Is there?

Reply:
I somehow missed that. updated it.